Add v5 migration guide and update changelog (#3658)
RevenueCat/purchases-ios: Sources/DocCDocumentation/DocCDocumentation.docc/V5_API_Migration_guide.md

    .build()
```

Additionally, because of the way in which StoreKit 2 operates, you must manually handle newly purchased transactions by calling `Purchases.shared.handleObserverModeTransaction()`. Subscription updates will be automatically tracked.

```swift
let result = try await product.purchase()
Purchases.shared.handleObserverModeTransaction(result)
```

If you're using observer mode with StoreKit 1, you will need to explicitly configure the SDK to use StoreKit 1:

```swift
Purchases.configure(with: .builder(withAPIKey: apiKey)
		.with(observerMode: true, storeKitVersion: .storeKit1)
    .build()
```

### Original Application Version

If you're converting a paid app to in-app subscriptions, and want to provide existing customers with certain features, [we recommend](https://www.revenuecat.com/blog/engineering/converting-a-paid-ios-app-to-subscriptions/) using the "original app version" field in customer info.

Because of a limitation of StoreKit 2, this field is not available if the customer is running on iOS 15, tvOS 15, macOS 12 or watchOS 8. If this is a requirement for you, we recommend switching back to StoreKit 1.

## Trusted Entitlements

> Version 5.0 of the SDK enables the Informational mode for Trusted Entitlements by default to protect your purchases from attackers. See the [Trusted Entitlements documentation](https://www.revenuecat.com/docs/trusted-entitlements) for more information.

Reviewer: I would rephrase this a bit since by default it doesn't actually "protect" you unless you check anything.
Maybe mention that informational will just log verification errors and allow you to check `customer info.entitlements.verification`.

## Deployment Target

The minimum targets have been raised to the folllowing:
- iOS 13.0
- tvOS 13.0
- watchOS 6.2
- macOS 10.15

## Release Assets

> Pre-built .frameworks will no longer be included in releases, only .xcframeworks: https://github.com/RevenueCat/purchases-ios/pull/3582

Reviewer: Maybe 
```suggestion
Pre-built `.frameworks` are no longer included in releases, only `.xcframeworks`: https://github.com/RevenueCat/purchases-ios/pull/3582
```
